File: Recursion/Python/fibonacci.py
def fibonacciIterative(n):
    a, b = 0, 1
    if n == 0:
        return a

    for i in range(2, n+1):
        c = a + b
        a = b
        b = c

    return b

def fibonacciRecursive(n):
    if n == 0 or n == 1:
        return n

    return fibonacciRecursive(n-1) + fibonacciRecursive(n-2)

if __name__ == '__main__':
    n = 10
    print("Fibonacci number at position", n, "(iterative) is:", fibonacciIterative(n))
    print("Fibonacci number at position", n, "(recursive) is:", fibonacciRecursive(n))

File: Recursion/C/fibonacci.c
#include <stdio.h>

int fibonacciIterative(int n) {
    int a = 0, b = 1, c, i;

    if(n == 0)
        return a;

    for(i = 2; i <= n; i++) {
        c = a + b;
        a = b;
        b = c;
    }
    return b;
}

int fibonacciRecursive(int n) {
    if(n == 0 || n == 1)
        return n;

    return fibonacciRecursive(n-1) + fibonacciRecursive(n-2);
}

int main() {
    int n = 10;
    printf("Fibonacci number at position %d (iterative) is: %d\n", n, fibonacciIterative(n));
    printf("Fibonacci number at position %d (recursive) is: %d\n", n, fibonacciRecursive(n));
    return 0;
}
